utils/highway-rewards-analysis/src/main.rs

```diff
@@ -0,0 +1,201 @@
+use std::{
+    collections::{BTreeMap, HashSet},
+    fs::File,
+    io::Read,
+};
+
+use clap::Parser;
+use flate2::read::GzDecoder;
+use serde::{Deserialize, Serialize};
+
+use casper_node::consensus::{
+    highway_core::{
+        finality_detector::{
+            assigned_weight_and_latest_unit, find_max_quora, round_participation,
+            RoundParticipation,
+        },
+        State,
+    },
+    protocols::common::validators,
+    utils::Validators,
+    ClContext,
+};
+use casper_types::{EraId, PublicKey, Timestamp, U512};
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    filename: String,
+    #[arg(short, long)]
+    verbose: bool,
+}
+
+/// Debug dump of era used for serialization.
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct EraDump {
+    /// The era that is being dumped.
+    pub id: EraId,
+
+    /// The scheduled starting time of this era.
+    pub start_time: Timestamp,
+    /// The height of this era's first block.
+    pub start_height: u64,
+
+    // omitted: pending blocks
+    /// Validators that have been faulty in any of the recent BONDED_ERAS switch blocks. This
+    /// includes `new_faulty`.
+    pub faulty: HashSet<PublicKey>,
+    /// Validators that are excluded from proposing new blocks.
+    pub cannot_propose: HashSet<PublicKey>,
+    /// Accusations collected in this era so far.
+    pub accusations: HashSet<PublicKey>,
+    /// The validator weights.
+    pub validators: BTreeMap<PublicKey, U512>,
+
+    /// The state of the highway instance associated with the era.
+    pub highway_state: State<ClContext>,
+}
+
+fn main() {
+    let args = Args::parse();
+
+    let mut data = vec![];
+    let mut file = File::open(&args.filename).unwrap();
+
+    if args.filename.ends_with(".gz") {
+        let mut gz = GzDecoder::new(file);
+        gz.read_to_end(&mut data).unwrap();
+    } else {
+        file.read_to_end(&mut data).unwrap();
+    }
+
+    let dump: EraDump = bincode::deserialize(&data).unwrap();
+
+    let validators =
+        validators::<ClContext>(&dump.faulty, &dump.cannot_propose, dump.validators.clone());
+
+    print_faults(&validators, &dump.highway_state);
+
+    print_skipped_rounds(&validators, &dump, args.verbose);
+
+    print_lowest_quorum_participation(&validators, &dump);
+}
+
+fn print_faults(validators: &Validators<PublicKey>, state: &State<ClContext>) {
+    if state.faulty_validators().count() == 0 {
+        return;
+    }
+
+    println!("Faults:");
+    for vid in state.faulty_validators() {
+        let fault = state.maybe_fault(vid).unwrap();
+        println!("{}: {:?}", validators.id(vid).unwrap(), fault);
+    }
+    println!();
+}
+
+const TOP_TO_PRINT: usize = 10;
```

Comment:
Maybe this should be an option?

Reply:
That would probably be a good idea for the future :+1: